racket/src/cs/io.sls

```diff
@@ -212,6 +212,25 @@
          (ftype-ref rktio_identity_t (a_bits) p)
          (ftype-ref rktio_identity_t (b_bits) p)
          (ftype-ref rktio_identity_t (c_bits) p))))
+
+    (define (rktio_date_to_vector p)
+      (let* ([p (make-ftype-pointer rktio_date_t (ptr->address p))]
+             [tzn (ftype-ref rktio_date_t (zone_name) p)])
+        (vector
+         (ftype-ref rktio_date_t (nanosecond) p)
+         (ftype-ref rktio_date_t (second) p)
+         (ftype-ref rktio_date_t (minute) p)
+         (ftype-ref rktio_date_t (hour) p)
+         (ftype-ref rktio_date_t (day) p)
+         (ftype-ref rktio_date_t (month) p)
+         (ftype-ref rktio_date_t (year) p)
+         (ftype-ref rktio_date_t (day_of_week) p)
+         (ftype-ref rktio_date_t (day_of_year) p)
+         (ftype-ref rktio_date_t (is_dst) p)
+         (ftype-ref rktio_date_t (zone_offset) p)
+         (if (eqv? tzn NULL)
+             #f
+             (address->ptr tzn)))))
```

Comment:
Is this intermediate vector needed, or would it be easy to go directly to a `date*` and avoid an extra allocation?